src/server/minecraft-server.js

```diff
@@ -0,0 +1,1503 @@
+                                'pipe'  // Direct child's stderr to parent stderr
+                            ]
+                        });
+                        
+                        starting = true;
+                        
+                        await this.checkForMinecraftToBeStarted();
+                        await this.updateStatus();
+                        await this.attachToMinecraft();
+                        await this.listCommands();
+                        await this.detachFromMinecraft();
+                        await this.log('Minecraft started.');
+                        return Promise.resolve();
+                    } catch (err) {
+                        return Promise.reject(err);
+                    }
+                } else {
+                    await this.log('Minecraft is already starting up.');
+                    return Promise.resolve('Minecraft is already starting up.');
+                }
+            } else {
+                await this.log(`Java not detected.`);
+                return Promise.reject(new Error(`Java not detected.`));
+            }
+        } else {
+            return Promise.reject(new Error('Minecraft is not installed.'));
+        }
+    }
+
+    /**
+     * Stops the Minecraft server process.
+     */
+    async stop () {
+        await this.log('Stopping MinecraftServer...');
+
+        let properties = this.properties;
+        let serverProcess = properties.serverProcess;
+        let started = properties.started;
+        let stopping = properties.stopping;
+
+        if (started) {
+            if (!stopping) {
+                stopping = true;
+                await this.attachToMinecraft();
+                serverProcess.stdin.write(`/stop${os.EOL}`);
+                await this.waitForBufferToBeFull();
+                await this.checkForMinecraftToBeStopped();
+                return Promise.resolve();
+            } else {
+                return Promise.reject(new Error('Minecraft is already shutting down.'));
+            }
+        } else {
+            stopping = false;
+            properties.stopped = true;
+            return Promise.resolve();
+        }
+    }
+
+    /**
+     * Reads the current state of files from the Minecraft server and gets
+     * player status if running.
+     */
+    async updateStatus () {
+        let properties = this.properties;
+        
+        await this.log('Updating Minecraft status.');
+        if (this.properties.installed) {
+            await this.checkForMinecraftUpdate();
+            await this.getEula();
+            await this.getServerProperties();
+            await this.getUserCache();
+            await this.getOps();
+            await this.getBannedIps();
+            await this.getBannedPlayers();
+            await this.getWhitelist();
+            if (properties.serverProcess && properties.serverProcess.pid) {
+                await this.listPlayers();
```

Comment:
This could be added to the above `Promise.all` call as
`properties.serverProcess && properties.serverProcess.pid ? this.listPlayers() : Promise.resolve()`